dbms/src/Processors/Formats/Impl/RegexpRowInputFormat.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <regex>
+#include <Core/Block.h>
+#include <Processors/Formats/IRowInputFormat.h>
+#include <Formats/FormatSettings.h>
+#include <Formats/FormatFactory.h>
+
+namespace DB
+{
+
+class ReadBuffer;
+
+
+class RegexpRowInputFormat : public IRowInputFormat
+{
+public:
+    RegexpRowInputFormat(ReadBuffer & in_, const Block & header_, Params params_,  const FormatSettings & format_settings_);
+
+    String getName() const override { return "RegexpRowInputFormat"; }
+
+    bool readRow(MutableColumns & columns, RowReadExtension & ext) override;
+
+private:
+    enum class FieldFormat
+    {
+        Escaped,
+        Quoted,
+        Csv,
+        Json,
+    };
+
+    bool readField(size_t index, MutableColumns & columns);
+    void readFieldsFromMatch(MutableColumns & columns, RowReadExtension & ext);
+    FieldFormat stringToFormat(const String & format);
+
+    const FormatSettings format_settings;
+    std::regex regexp;
```

Comment:
It's better to use `re2` instead of `std::regex` for performance reasons. Also it will be consistent with other places in ClickHouse where `re2` is used.